### DISCUSSION TITLE
pixi add error - regex constraints are not supported

### DISCUSSION THREAD

**Discussion by noamgot** [prefix-dev/pixi#2709] [Q&A] [answered]
Hello there
I'm not sure it's a bug on pixi's side, so before I open an actual bug issue I decided to ask here...

### background
I try to migrate one of my work repos to use pixi (our usual workflow is based on conda environments, no actual package management is applied). 

### The problem
I tried migrating directly by running `pixi init --import environment.yml`, and when I tried installing an environment I got an error with a specific  (internal) package, let's call it here "foo"
```
  > pixi shell
  × failed to solve the conda requirements of 'default' 'linux-64'
  ╰─▶ Cannot solve the request because of: The following packages are incompatible
      └─ foo >3,<4 cannot be installed because there are no viable options:
         └─ foo 3.0.1 | 3.0.2 | 3.0.3 | 3.0.4 | 3.0.5 | 3.0.6 | 3.0.7 | 3.0.8 | 3.0.9 | 3.0.10 | 3.0.11 is excluded because the dependency 'opensearch-py >=2.*.*' failed to parse: invalid version
      constraint: regex constraints are not supported

```

I also created a fresh project with `pixi init` , and when I tried `pixi add "foo>3,<4"`  I got the same failure. 
For completeness, I also tested directly installing a package with a regex constraint, that fails too:
```
❯ pixi add "opensearch-py>=2.*.*"               
  × invalid version constraint: regex constraints are not supported
  ╰─▶ regex constraints are not supported
```

**Important**: I do manage to create a fresh conda environment with the exact same package as a single dependency!
```
mamba create -n foo-env "foo>3<4"
mamba create -n bar "opensearch-py>=2.*.*" 
# both works!
```

I googled for "regex constraints are not supported" and found nothing. I can't tell whether it's a problem with pixi or maybe the package solver (or something else), if you have any lead on this it would be great.
thanks

**Comment by noamgot** [comment]
ping 🏓

**Comment by baszalmstra** [comment, marked as answer]
You have ran into a genuine issue that I think we should support! This is an issue with our version spec parsing (in rattler). We always assume a an optional operator followed by a version or a version followed by .*. However your case is neither, but it is something we should support.

**Reply by noamgot** [reply]
thanks! should I open an issue on rattler's repo with this one?

**Reply by baszalmstra** [reply]
that would be great!

**Reply by noamgot** [reply]
done: https://github.com/conda/rattler/issues/1004

**Comment by noamgot** [comment]
So this issue is fixed on rattler's side, see https://github.com/conda/rattler/pull/1006 (thanks!)

When can we expect to see this change in pixi?

**Reply by lucascolley** [reply]
this is working now:

```
❯ pixi add "python>3,<4"
✔ Added python>3,<4

❯ pixi add "opensearch-py>=2.*.*"
✔ Added opensearch-py>=2.*.*
```
